module/data/item/race.mjs

```diff
@@ -0,0 +1,73 @@
+import SystemDataModel from "../abstract.mjs";
+import { AdvancementField, IdentifierField } from "../fields.mjs";
+import ItemDescriptionTemplate from "./templates/item-description.mjs";
+
+/**
+ * Data definition for Race items.
+ * @mixes ItemDescriptionTemplate
+ *
+ * @property {string} identifier       Identifier slug for this race.
+ * @property {object[]} advancement    Advancement objects for this race.
+ * @property {number} darkvision       Darkvision distance.
+ * @property {object} movement
+ * @property {number} movement.burrow
+ * @property {number} movement.climb
+ * @property {number} movement.fly
+ * @property {number} movement.swim
+ * @property {number} movement.walk
+ * @property {string} movement.units   Units used to measure movement.
+ * @property {object} type
+ * @property {string} type.value       Creature type (e.g. "humanoid", "construct").
+ * @property {string} type.subtype     Additional type data (e.g. "elf" for Eladrin)
+ */
+export default class RaceData extends SystemDataModel.mixin(ItemDescriptionTemplate) {
+  /** @inheritdoc */
+  static defineSchema() {
+    return this.mergeSchema(super.defineSchema(), {
+      identifier: new IdentifierField({label: "DND5E.Identifier"}),
+      advancement: new foundry.data.fields.ArrayField(new AdvancementField(), {label: "DND5E.AdvancementTitle"}),
+      darkvision: new foundry.data.fields.NumberField({min: 0, integer: true, label: "DND5E.SenseDarkvision"}),
+      movement: new foundry.data.fields.SchemaField({
+        burrow: new foundry.data.fields.NumberField({
+          initial: null, min: 0, integer: true, label: "DND5E.MovementBurrow"
+        }),
+        climb: new foundry.data.fields.NumberField({
+          initial: null, min: 0, integer: true, label: "DND5E.MovementClimb"
+        }),
+        fly: new foundry.data.fields.NumberField({
+          initial: null, min: 0, integer: true, label: "DND5E.MovementFly"
+        }),
+        swim: new foundry.data.fields.NumberField({
+          initial: null, min: 0, integer: true, label: "DND5E.MovementSwim"
+        }),
+        walk: new foundry.data.fields.NumberField({
+          initial: 30, min: 0, integer: true, label: "DND5E.MovementWalk"
+        }),
+        units: new foundry.data.fields.StringField({initial: "ft", label: "DND5E.MovementUnits"})
+      }),
+      type: new foundry.data.fields.SchemaField({
+        value: new foundry.data.fields.StringField({initial: "humanoid", label: "DND5E.CreatureType"}),
```

Comment:
`type.value` field should specify a valid set of `choices` I think. While we're here, do we like `value` as this attribute path? I know we've used it before, but maybe something like `type.primary` or `type.category` would be more accurate?

Reply:
There's unfortunately a bit of tension with regards to putting explicit `choices` in the DataModel while still allowing the system to be configurable. If we set `choices` to something like `CONFIG.creatureTypes`, and a module adds something to that, any Actors that are subsequently configured with that creature type will become invalid if the module is ever disabled.

There are a few things we've thought about doing here: Deferring to preparation to strip out 'unconfigured' values, or writing custom validators to do something similar. At the moment we've opted to just not use `choices` for now, until we either implement some of the aforementioned options, or there is some core API solution in place.

Reply:
The problem with setting a `choices` is it easily leads to entire actors/items being invalid. If a module or script adds a choice and is then disabled, any actors or items that use that choice will just disappear entirely from the world and the only way a user can know what went wrong is through the inspector. I'd prefer to avoid `choices` and just degrade more gracefully when creating labels and whatnot.

Renaming would work, but would we also want to migrate the NPC data structure to match?